internal/pkg/service/common/etcdop/op/atomic.go

```diff
@@ -279,6 +295,20 @@ func (v *AtomicOp[R]) DoWithoutRetry(ctx context.Context, opts ...Option) *TxnRe
 					RangeEnd:    op.RangeEnd, // may be empty
 				},
 			)
+
+			// See SkipPrefixKeysCheck method documentation, by default, the feature is enabled.
+			if v.checkPrefixKey {
+				if op.RangeEnd != nil {
+					for _, kv := range op.KVs {
+						cmps = append(cmps, etcd.Cmp{
+							Target:      etcdserverpb.Compare_MOD,
+							Result:      etcdserverpb.Compare_GREATER,
+							TargetUnion: &etcdserverpb.Compare_ModRevision{ModRevision: 0},
+							Key:         kv.Key,
+						})
+					}
+				}
+			}
```

Comment:
Implementation.

`op.RangeEnd != nil` it is a Get operation over range/prefix, not a single key get.